torch/utils/data/_utils/collate.py

```diff
@@ -75,6 +75,10 @@ def default_collate(batch):
     elif isinstance(elem, tuple) and hasattr(elem, '_fields'):  # namedtuple
         return elem_type(*(default_collate(samples) for samples in zip(*batch)))
     elif isinstance(elem, container_abcs.Sequence):
+        it = iter(batch)
+        elem_size = len(next(it))
+        assert all(len(elem) == elem_size for elem in it), \
```

Comment:
Thanks for updating with the test! One final small thing. Generally we don't really throw `AssertionError` (unless it is violation of an invariance which signals a PyTorch bug). Could you change this to throw `RuntimeError` please? Thanks!

Reply:
Wow, that was a quick review. Addressed in newest update!